src/aws-cpp-sdk-core/source/auth/signer/AWSAuthV4Signer.cpp

```diff
@@ -264,6 +265,7 @@ bool AWSAuthV4Signer::SignRequest(Aws::Http::HttpRequest& request, const char* r
         {
             payloadHash = STREAMING_UNSIGNED_PAYLOAD_TRAILER;
             Aws::String trailerHeaderValue = Aws::String("x-amz-checksum-") + request.GetRequestHash().first;
+            request.DeleteHeader(trailerHeaderValue.c_str());
```

Comment:
why is trailer being deleted?

Reply:
not a good name for the variable re-naming to "checksumHeaderValue". We are removing the checksum header from being sent as it is part of the trailer in this branch of logic. The bug beforehand was that a user would add the checksum to the request, it would serialized as a header AND calculated as part of the request. thus my comment on the [original PR](https://github.com/aws/aws-sdk-cpp/pull/2644)

> we should consider not serializing the header x-amz-checksum- to the header in the first place during codegen as we will always end up adding it one way or another if needed. this is fine for now imo opinon, but removing the serialization could perhaps be a better long term solution

so when sent there would be a header and a trailer, which is wrong.